changefeedccl: clear schema registry singleton upon starting tests

samiskin opened the pull request:
Resolves https://github.com/cockroachdb/cockroach/issues/102671
Resolves https://github.com/cockroachdb/cockroach/issues/102853
Resolves https://github.com/cockroachdb/cockroach/issues/102836
Resolves https://github.com/cockroachdb/cockroach/issues/102867
Resolves https://github.com/cockroachdb/cockroach/issues/102832

Since the schemaRegistrySingleton would persist across tests, it may be set by one test, then that test would close its server freeing up that port, then another test may get the same port and have to deal with a polluted cache.

We should still deal with this kind of case of a polluted cache in the product as well for the edge case where a user messes with their schema registry while we've cached information about it, but it feels still wrong to have all unit tests running with the unpredictable effects of other unit tests.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/104538)
<!-- Reviewable:end -->

samiskin committed:
changefeedccl: clear schema registry singleton upon starting tests

Since the schemaRegistrySingleton would persist across tests, it may be
set by one test, then that test would close its server freeing up that
port, then another test may get the same port and have to deal with a
polluted cache.

We should still deal with this kind of case of a polluted cache in the
product as well for the edge case where a user messes with their schema
registry while we've cached information about it, but it feels still
wrong to have all unit tests running with the unpredictable effects of
other unit tests.

Release note: None

miretskiy commented:
blathers backport 22.3 22.2
